server/FakeToOriginalIPConverter/FakeToOriginalIPConverter.js

```diff
@@ -49,28 +49,36 @@ async function modifyClients() {
     let applicationList = fakeToOriginalIPMapping['fake-to-original-iP-mapping'];
     for (let index = 0; index < applicationList.length; index++) {
         let application = applicationList[index];
-        let applicationName = application['component'];
-        let releaseNumber = application['release'];
-        let originalTcpIpAddress = application['original-address'];
-        let originalTcpPort = application['original-tcp-port'];
-        let httpClientUuid = await httpClientInterface.getHttpClientUuidAsync(applicationName, releaseNumber);
-        if (httpClientUuid != undefined) {
-            const tcpClientuuidList = await LogicalTerminationPoint.getServerLtpListAsync(httpClientUuid);
-            if (tcpClientuuidList != undefined) {
-                // need to update after including the NGINX concept
-                let tcpClientuuid = tcpClientuuidList[0];
-                let isTcpIPAddressUpdated = false;
-                let isTcpPortUpdated = false;
-                try {
-                    isTcpIPAddressUpdated = await tcpClientInterface.setRemoteAddressAsync(tcpClientuuid, originalTcpIpAddress);
-                    isTcpPortUpdated = await tcpClientInterface.setRemotePortAsync(tcpClientuuid, originalTcpPort);
-                } catch (error) {
-                    console.log(error);
+        if(application['instance-type'] == 'client'){
+            let applicationName = application['component'];
+            let releaseNumber = application['release'];
+            let originalTcpIpAddress = application['original-address'];
+            let originalTcpPort = application['original-tcp-port'];
+            let originalTcpProtocol = application['protocol'];
+            let originalReleaseNumber = application['new-release'];
+            let httpClientUuid = await httpClientInterface.getHttpClientUuidAsync(applicationName, releaseNumber);
+            if (httpClientUuid != undefined) {
```

Comment:
For the client , I could able to see that , we added a new attribute called “new-release”.
And the existing attribute “release” carries the details of the old-release.

You are trying to find the ltp instance for the old-release and then updating the 
- old-release to new release 
- old-ip-address, port , protocol to new

**For example ,** 
In the fakeIPAddress mapping file , lets say for the EATL application , the details are as below ,
old-release = 1.0.0 , new-release = 2.0.0 , ip-address = 1.1.1.1 , port = 3000 , protocol = https
In the load file , if EATL client is in the old-release 1.0.0 , then the new-release and tcp information shall be updated.
**Problem :**
But , what if ? the load file already consists of the new-release 2.0.0 ? 
Looks like the fake-ips will not be replaced for this instance.

**Can we modify the script as below ?**
**Assumption :**
fakeIPAddress mapping file always consists of the updated version that is running in the production.
That means , if live network consists of the version 33.0.0 , then only that information will be available in this file.
**change :**

```
Find the ltp instance based on the application-name and new-release number
    If instance found then 
    Update the tcp information 
Else 
    Find the ltp instance based on the application-name and old-release number
    If instance found then 
        Update the release information 
        Update the tcp information 
```